aopy/visualization/base.py

```diff
@@ -1425,8 +1425,143 @@ def plot_waveforms(waveforms, samplerate, plot_mean=True, ax=None):
     else:
         ax.plot(time_axis, waveforms)
 
-    ax.set_xlabel(r'Time ($\mu$s)')
+    ax.set_xlabel(r'Time ($\mu$s)')    
+    
+def plot_condition_tuning(per_condition_data, conditions, ylabel='success rate', ax=None, **kwargs):
+    '''
+    Plot tuning curves for categorical data. Essentially a scatter plot with the mean of each condition
+    plotted as a solid line.
+
+    Args:
+        per_condition_data (nconditions, ...): data for each condition
+        conditions (nconditions): condition for each data point
+        ylabel (str, optional): label for the y-axis. Default "success rate"
+        ax (pyplot.Axes, optional): axis to plot the tuning curves on. Default the current axis.
+
+    Examples:
+
+        Plot the success rate for 4 different conditions
+
+        .. code-block:: python
+            direction = [-np.pi, -np.pi/2, 0, np.pi/2]
+            data = np.random.normal(0, 1, (4, 2, 4))
+            
+            fig = plt.figure()
+            plot_condition_tuning(data, np.degrees(direction))
+            
+        .. image:: _images/condition_tuning.png
+    '''
+    if ax is None:
+        ax = plt.gca()
+
+    per_condition_data = np.array(per_condition_data).reshape(len(per_condition_data), -1)
+    ntr = per_condition_data.shape[1]
+    print(per_condition_data.shape)
+    print(conditions.shape)
+
+    # Scatter plot
+    plt.scatter(np.tile(conditions, (1,ntr)), per_condition_data, **kwargs)
+
+    # Add means
+    dist = (np.max(conditions) - np.min(conditions)) / len(conditions)
+    for idx, cond in enumerate(per_condition_data):
+        mean = np.mean(cond)
+        ax.plot([conditions[idx]-dist/2, conditions[idx]+dist/2], [mean, mean], 'r-')
+
+    ax.set_xlabel('condition')
+    ax.set_ylabel(ylabel)
+
+def plot_direction_tuning(per_direction_data, directions, show_var=True, wrap=True, ylabel='success rate', ax=None):
+    '''
+    Plot tuning curves for directional data. The mean is plotted as a solid line and the variance as 
+    a shaded region around the mean. Works with both cartesian and polar axes.
+    
+    Args:
+        mean (ndir, nch, ntrial): direction responses for each channel. If only one channel, can be (ndir, ntrial).
+        directions (ndir): unique directions in radians
+        show_var (bool, optional): if True, shows the variance around the mean. Default True.
```

Comment:
Might be beyond the scope of this pull request but it would be nice for the user to be able to input their own error. For example, if they wanted to show SD instead of variance.

Reply:
maybe someone else can do that :)